warehouse/ingest-core/src/main/java/datawave/ingest/data/TypeRegistry.java

```diff
@@ -60,13 +60,9 @@ public class TypeRegistry extends HashMap<String,Type> {
     
     private static Multimap<String,String> handlerExpansions = HashMultimap.create();
     
-    public static TypeRegistry getInstance(Configuration config) {
+    public synchronized static TypeRegistry getInstance(Configuration config) {
         if (null == registry) {
-            synchronized (lock) {
-                if (null == registry) {
-                    registry = new TypeRegistry(config);
-                }
-            }
+            registry = new TypeRegistry(config);
         }
         return registry;
     }
```

Comment:
It would probably be more efficient to leave this code as-is.  In the original code, if the registry is not null, then access to the registry resource will never be blocked since the synchronize call is skipped.  With these changes, only one thread will be able to access the registry resource at a time, blocking all other threads.

Reply:
Did some more reading on this.  Seems like we can just mark the resource `volatile` and it should behave appropriately with the existing code. (Same with other spots.)